File: Classes/GameMap/GameMap.py
from Classes.GameMap.FightPoint import FightPoint

class GameMap:
    def __init__(self, image, fightPoints, pointDistTollerance):
        """
        This class creates a new game map.
        :param image: Full image of the map
        :param fightPoints: list of FightPoint objects
        :param pointDistTollerance: float (used to calculate de difference between te current point and the other
         fighPoint objects)
        """
        self._currentPoint = FightPoint(fightPoints[0].getPoint(), fightPoints[0].getAllowedDirections(),
                                        fightPoints[0].getActivated())

        self._previousPoint = FightPoint(fightPoints[0].getPoint(), fightPoints[0].getAllowedDirections(),
                                        fightPoints[0].getActivated())
        self._image = image
        self._fightPoints = fightPoints
        self.tol = pointDistTollerance

    def pointMatchTest(self, point):
        """
        :param point: FightPoint object
        :return: boolean
        """
        return (point.getPoint()[0] - self.getTol() <= self.getCurrentPoint().getPoint()[0] <= point.getPoint()[0] + self.getTol() and
                point.getPoint()[1] - self.getTol() <= self.getCurrentPoint().getPoint()[1] <= point.getPoint()[1] + self.getTol())

    def pointMatch(self, point):
        test = self.pointMatchTest(point)
        if test:
            self.getCurrentPoint().setPoint(point.getPoint())
            self.getCurrentPoint().setAllowedDirections(point.getAllowedDirections())
            self.getCurrentPoint().setActivated(point.getActivated())
        return test

    def fightPointTest(self):
        list = self.getFightPoints()

        for fighPoint in list:
            if self.pointMatch(fighPoint):
                return True

        return False

    def disableCurrentPoint(self):
        list = self.getFightPoints()

        for fightPoint in list:
            if self.pointMatchTest(fightPoint):
                fightPoint.deactivate()
                self.getCurrentPoint().deactivate()


    def getCurrentPoint(self):
        return self._currentPoint

    def setCurrentPoint(self, point):
        self.getCurrentPoint().setPoint(point.getPoint())
        self.getCurrentPoint().setAllowedDirections(point.getAllowedDirections())
        self.getCurrentPoint().setActivated(point.getActivated())

    def getImage(self):
        return self._image

    def setImage(self, image):
        self._image = image

    def getFightPoints(self):
        return self._fightPoints

    def getTol(self):
        return self.tol

    def setTol(self, tol):
        self.tol = tol

    def getPreviousPoint(self):
        return self._previousPoint

    def setPreviousPoint(self, point):
        self.getPreviousPoint().setPoint(point.getPoint())
        self.getPreviousPoint().setAllowedDirections(point.getAllowedDirections())
        self.getPreviousPoint().setActivated(point.getActivated())

File: TheFlameRealmJava/core/src/main/java/com/flamerealm/game/gamemap/GameMap.java
package com.flamerealm.game.gamemap;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.math.Vector2;

import java.util.List;

/**
 * Port de GameMap.py. As posicoes sempre sao substituidas por um Vector2 novo
 * (nunca mutadas in-place via .set()), do mesmo jeito que o Python original
 * sempre substituia a tupla inteira - assim currentPoint/previousPoint nunca
 * corrompem os Vector2 constantes compartilhados de GameConstants.
 */
public class GameMap {
    private final FightPoint currentPoint;
    private final FightPoint previousPoint;
    private Texture image;
    private final List<FightPoint> fightPoints;
    private float tol;

    public GameMap(Texture image, List<FightPoint> fightPoints, float pointDistTollerance) {
        FightPoint first = fightPoints.get(0);
        this.currentPoint = new FightPoint(new Vector2(first.getPoint()), first.getAllowedDirections(), first.getActivated());
        this.previousPoint = new FightPoint(new Vector2(first.getPoint()), first.getAllowedDirections(), first.getActivated());
        this.image = image;
        this.fightPoints = fightPoints;
        this.tol = pointDistTollerance;
    }

    public boolean pointMatchTest(FightPoint point) {
        return point.getPoint().x - getTol() <= getCurrentPoint().getPoint().x
                && getCurrentPoint().getPoint().x <= point.getPoint().x + getTol()
                && point.getPoint().y - getTol() <= getCurrentPoint().getPoint().y
                && getCurrentPoint().getPoint().y <= point.getPoint().y + getTol();
    }

    public boolean pointMatch(FightPoint point) {
        boolean test = pointMatchTest(point);
        if (test) {
            getCurrentPoint().setPoint(point.getPoint());
            getCurrentPoint().setAllowedDirections(point.getAllowedDirections());
            getCurrentPoint().setActivated(point.getActivated());
        }
        return test;
    }

    public boolean fightPointTest() {
        for (FightPoint fightPoint : getFightPoints()) {
            if (pointMatch(fightPoint)) {
                return true;
            }
        }
        return false;
    }

    public void disableCurrentPoint() {
        for (FightPoint fightPoint : getFightPoints()) {
            if (pointMatchTest(fightPoint)) {
                fightPoint.deactivate();
                getCurrentPoint().deactivate();
            }
        }
    }

    public FightPoint getCurrentPoint() {
        return currentPoint;
    }

    public void setCurrentPoint(FightPoint point) {
        getCurrentPoint().setPoint(point.getPoint());
        getCurrentPoint().setAllowedDirections(point.getAllowedDirections());
        getCurrentPoint().setActivated(point.getActivated());
    }

    public Texture getImage() {
        return image;
    }

    public void setImage(Texture image) {
        this.image = image;
    }

    public List<FightPoint> getFightPoints() {
        return fightPoints;
    }

    public float getTol() {
        return tol;
    }

    public void setTol(float tol) {
        this.tol = tol;
    }

    public FightPoint getPreviousPoint() {
        return previousPoint;
    }

    public void setPreviousPoint(FightPoint point) {
        getPreviousPoint().setPoint(point.getPoint());
        getPreviousPoint().setAllowedDirections(point.getAllowedDirections());
        getPreviousPoint().setActivated(point.getActivated());
    }
}
